From 35815463599b2aa843edcc976920d996f6e4b0e2 Mon Sep 17 00:00:00 2001
From: dol-sen <brian.dolbec@gmail.com>
Date: Wed, 27 Apr 2011 03:58:06 -0700
Subject: [PATCH] fix broken *args for python 2.5 and 2.6+.
 Also fixes my previous fix & resultant breakage.
 Passes tests for all python versions supported.

---
 bin/layman                   |    2 +-
 layman/overlays/bzr.py       |    2 +-
 layman/overlays/darcs.py     |    2 +-
 layman/overlays/git.py       |    2 +-
 layman/overlays/mercurial.py |    2 +-
 layman/overlays/rsync.py     |    2 +-
 layman/overlays/source.py    |    4 ++--
 layman/overlays/svn.py       |    4 ++--
 layman/overlays/tar.py       |    2 +-
 9 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/bin/layman b/bin/layman
index b85f4fa..01ee0c8 100755
--- a/bin/layman
+++ b/bin/layman
@@ -1,4 +1,4 @@
-#!python
+#!/usr/bin/python
 ################################################################################
 # LAYMAN - A UTILITY TO SELECT AND UPDATE GENTOO OVERLAYS
 ################################################################################
diff --git a/layman/overlays/bzr.py b/layman/overlays/bzr.py
index 4725c98..b59cf6c 100644
--- a/layman/overlays/bzr.py
+++ b/layman/overlays/bzr.py
@@ -52,7 +52,7 @@ class BzrOverlay(OverlaySource):
 
         # bzr get SOURCE TARGET
         args = ['get', self.src + '/', path([base, self.parent.name])]
-        return self.run_command(*args)
+        return self.run_command(args)
 
     def sync(self, base, quiet = False):
         '''Sync overlay.'''
diff --git a/layman/overlays/darcs.py b/layman/overlays/darcs.py
index 12e75ec..c062543 100644
--- a/layman/overlays/darcs.py
+++ b/layman/overlays/darcs.py
@@ -51,7 +51,7 @@ class DarcsOverlay(OverlaySource):
 
         # darcs get --partial SOURCE TARGET
         args = ['get', '--partial', self.src + '/', path([base, self.parent.name])]
-        return self.run_command(*args)
+        return self.run_command(args)
 
     def sync(self, base, quiet = False):
         '''Sync overlay.'''
diff --git a/layman/overlays/git.py b/layman/overlays/git.py
index 8db6c51..3b2608b 100644
--- a/layman/overlays/git.py
+++ b/layman/overlays/git.py
@@ -60,7 +60,7 @@ class GitOverlay(OverlaySource):
             args.append('-q')
         args.append(fix_git_source(self.src))
         args.append(path([base, self.parent.name]))
-        return self.run_command(*args)
+        return self.run_command(args)
 
     def sync(self, base, quiet = False):
         '''Sync overlay.'''
diff --git a/layman/overlays/mercurial.py b/layman/overlays/mercurial.py
index ea3f9da..8bf919b 100644
--- a/layman/overlays/mercurial.py
+++ b/layman/overlays/mercurial.py
@@ -51,7 +51,7 @@ class MercurialOverlay(OverlaySource):
 
         # hg clone SOURCE TARGET
         args = ['clone', self.src + '/', path([base, self.parent.name])]
-        return self.run_command(*args)
+        return self.run_command(args)
 
     def sync(self, base, quiet = False):
         '''Sync overlay.'''
diff --git a/layman/overlays/rsync.py b/layman/overlays/rsync.py
index 3a5a342..bf01462 100644
--- a/layman/overlays/rsync.py
+++ b/layman/overlays/rsync.py
@@ -66,7 +66,7 @@ class RsyncOverlay(OverlaySource):
         args.append(self.src + '/')
         args.append(path([base, self.parent.name]))
 
-        return self.run_command(*args)
+        return self.run_command(args)
 
     def supported(self):
         '''Overlay type supported?'''
diff --git a/layman/overlays/source.py b/layman/overlays/source.py
index 2c039c9..9e7f974 100644
--- a/layman/overlays/source.py
+++ b/layman/overlays/source.py
@@ -107,9 +107,9 @@ class OverlaySource(object):
     def command(self):
         return self.config['%s_command' % self.__class__.type_key]
 
-    def run_command(self, *args, **kwargs):
+    def run_command(self, args, **kwargs):
         file_to_run = _resolve_command(self.command())[1]
-        args = (file_to_run, ) + args
+        args = [file_to_run] + args
         assert('pwd' not in kwargs)  # Bug detector
 
         cwd = kwargs.get('cwd', None)
diff --git a/layman/overlays/svn.py b/layman/overlays/svn.py
index 2a30198..092d786 100644
--- a/layman/overlays/svn.py
+++ b/layman/overlays/svn.py
@@ -56,7 +56,7 @@ class SvnOverlay(OverlaySource):
         args.append(self.src + '/@')
         args.append(path([base, self.parent.name]))
 
-        return self.run_command(*args)
+        return self.run_command(args)
 
     def sync(self, base, quiet = False):
         '''Sync overlay.'''
@@ -77,7 +77,7 @@ class SvnOverlay(OverlaySource):
             args.append('-q')
         args.append(checkout_location())
 
-        return self.run_command(*args)
+        return self.run_command(args)
 
     def supported(self):
         '''Overlay type supported?'''
diff --git a/layman/overlays/tar.py b/layman/overlays/tar.py
index 919f4df..612b6b6 100644
--- a/layman/overlays/tar.py
+++ b/layman/overlays/tar.py
@@ -128,7 +128,7 @@ class TarOverlay(OverlaySource):
 
         # tar -v -x -f SOURCE -C TARGET
         args = ['-v', '-x', '-f', pkg, '-C', dest_dir]
-        result = self.run_command(*args)
+        result = self.run_command(args)
 
         os.unlink(pkg)
         return result
-- 
1.7.3.4

